payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/PostalCodeConfig.kt

```diff
@@ -0,0 +1,80 @@
+package com.stripe.android.ui.core.elements
+
+import androidx.annotation.StringRes
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.VisualTransformation
+import kotlinx.coroutines.flow.MutableStateFlow
+
+internal class PostalCodeConfig(
+    @StringRes override val label: Int,
+    override val capitalization: KeyboardCapitalization = KeyboardCapitalization.Words,
+    override val keyboard: KeyboardType = KeyboardType.Text,
+    override val trailingIcon: MutableStateFlow<TextFieldIcon?> = MutableStateFlow(null),
+    country: String
+) : TextFieldConfig {
+    override val debugLabel: String = "postal_code_text"
+    override val visualTransformation: VisualTransformation? = null
+    override val loading: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    private val format = CountryPostalFormat.forCountry(country)
+
+    override fun determineState(input: String): TextFieldState = object : TextFieldState {
+        override fun shouldShowError(hasFocus: Boolean) = false
+
+        override fun isValid(): Boolean {
+            return input.length in format.minimumLength..format.maximumLength &&
+                input.matches(format.regexPattern)
+        }
+
+        override fun getError(): FieldError? = null
+
+        override fun isFull(): Boolean = input.length >= format.minimumLength
+
+        override fun isBlank(): Boolean = input.isBlank()
+    }
+
+    override fun filter(userTyped: String): String =
+        if (
+            setOf(KeyboardType.Number, KeyboardType.NumberPassword).contains(keyboard)
+        ) {
+            userTyped.filter { it.isDigit() }
+        } else {
+            userTyped
+        }.replace(Regex("\\s*"), "")
+
+    override fun convertToRaw(displayName: String) = displayName
+
+    override fun convertFromRaw(rawValue: String) = rawValue
+
+    sealed class CountryPostalFormat(
+        val minimumLength: Int,
+        val maximumLength: Int,
+        val regexPattern: Regex
+    ) {
+        object CA : CountryPostalFormat(
+            minimumLength = 6,
+            maximumLength = 6,
+            regexPattern = Regex("([a-zA-Z]\\d)+")
```

Comment:
Should we allow spaces for Canadian postal codes? Currently, it’s not possible to enter them.

Reply:
And another 🇨🇦 question: Should we auto-capitalize the postal code? I find `a1b 2c3` looks odd, and `A1B 2C3` looks much nicer.

Reply:
Added spacing regex, won't fix the auto-capitalization